pyEpiabm/pyEpiabm/intervention/case_isolation.py

```diff
@@ -40,13 +42,26 @@ def __call__(self, time):
                         # Stop isolating people after their isolation period
                         person.isolation_start_time = None
                 else:
-                    if person.is_symptomatic():
+                    if self.person_selection_method(person):
                         r = random.random()
                         # Require symptomatic individuals to self-isolate
                         # with given probability
                         if r < self.isolation_probability:
                             person.isolation_start_time = time + self.\
                                                           isolation_delay
+                            if person.date_positive is not None:
+                                self._population.test_isolate_count = [0, 0]
+                                if person.is_symptomatic():
+                                    self._population.test_isolate_count[0] += 1
+                                else:
+                                    self._population.test_isolate_count[1] += 1
+
+    def person_selection_method(self, person):
```

Comment:
What is this selection for? Can we add a docstring?

Reply:
> What is this selection for? Can we add a docstring?

Good, just make sure to add a full docstring as this is in the main source code (so include a record of parameters/outputs)